### DISCUSSION TITLE
VPN setup for TPP Level 4 access requires a native windows environment

### DISCUSSION THREAD

**Discussion by venexia** [opensafely/documentation#251] [Installing OpenSAFELY] [answered]
The VPN setup for TPP Level 4 access requires a native windows environment to run the SonicWall VPN software. For Mac users, the TPP technician recommended parallels to access a virtual machine but this is a paid-for software option. Does anyone know of a free alternative? 

I am aware that you can use Boot Camp Assitant to partition your mac and run native windows but this requires 64GB or more free storage space on your Mac startup disk, which I currently do not have. Also, remote access to a windows machine is not an option here because the connection to the remote desktop will be stopped when the VPN connects.

**Comment by ghickman** [comment, marked as answer]
I've been using VMware Fusion, which is free for non-comercial uses, which I believe OpenSAFELY comes under.  I think you might have to create an account with VMware to sign up for the free license.

I've been using one of the free Windows 10 ISO images with it, some of them expire after a few months so you have to re-install.

**Reply by venexia** [reply]
Great, thank you!

For the benefit of future users:
- Sign up for a VMWare account and download VMWare fusion [here](https://www.vmware.com/uk/products/fusion.html)
- Download a Windows disc image [here](https://www.microsoft.com/en-us/software-download/windows10ISO)
- Follow the instructions from [this video](https://www.youtube.com/watch?v=xXaeYGYAM_0) to complete the setup

**Comment by elsie-h** [comment]
@ghickman 

I'm running into the same issue, but as I have a Mac with an Apple M1 chip, the above answer does not solve the issue.

The university were happy to purchase a Parallels licence for me as currently Parallels seems to be [the only option for running Windows on an M1 Mac](https://machow2.com/windows-on-m1-mac/) (from the comments on [this article](https://blogs.vmware.com/teamfusion/2021/09/fusion-for-m1-public-tech-preview-now-available.html), the preview version of VMWare doesn't seem to support Windows).

So far, I've installed Windows 11 using Parallels, and attempted to install the Sonicwall VPN Client, as outlined in OpenSAFELY Level 4 Information / Installation Guide. However, I got the following error:

![Screenshot 2021-12-06 at 17 46 47](https://user-images.githubusercontent.com/31345323/145080658-c764769b-e535-435b-898b-9c365d0c8e3d.png)

While I didn't find anything particularly helpful when googling the error, [this link](https://machow2.com/windows-on-m1-mac/) does state that many programs and applications do not work with Windows for ARM, so maybe this is the issue? With my limited computing knowledge I can't be sure this is the same issue, but it seems to suggest [here](https://www.sonicwall.com/support/knowledge-base/do-connect-tunnel-or-modern-connect-tunnel-support-arm-based-processors/200505220944370/) that Sonicwall clients are not supported for ARM-based devices.

I'm going to have a go with Windows 10 too when I get a chance. However, if anyone else has found a solution already, please let me know!

Just to note, @venexia now has a Mac with an Apple M1 chip, so is also running into these issues. She was possibly going to give VMWare a go, but I'm sure will post here if any progress!

**Reply by StevenMaude** [reply]
@elsie-h: Someone else actually reported the same problem to us a few weeks ago. I'm not sure if there's any known workaround as yet.

I opened https://github.com/opensafely/documentation/issues/512 to at least document this, that would hopefully save anyone else's time in testing this particular setup out.
